ecosystem/typescript/sdk/src/transactions/tests/async_queue.test.ts

```diff
@@ -0,0 +1,80 @@
+import { AsyncQueue, AsyncQueueCancelledError } from "../async_queue";
+
+describe("asyncQueue", () => {
+  it("should enqueue and dequeue items as a normal queue", async () => {
+    const asyncQueue = new AsyncQueue<number>();
+
+    asyncQueue.enqueue(1);
+
+    const item1 = await asyncQueue.dequeue();
+
+    asyncQueue.enqueue(2);
+    asyncQueue.enqueue(3);
+
+    const item2 = await asyncQueue.dequeue();
+    const item3 = await asyncQueue.dequeue();
+
+    expect(item1).toBe(1);
+    expect(item2).toBe(2);
+    expect(item3).toBe(3);
+  });
+
+  it("should handle dequeue before queue", async () => {
+    const asyncQueue = new AsyncQueue<number>();
+
+    const itemPromise1 = asyncQueue.dequeue();
+    const itemPromise2 = asyncQueue.dequeue();
+
+    asyncQueue.enqueue(1);
+    asyncQueue.enqueue(2);
+
+    const item1 = await itemPromise1;
+    const item2 = await itemPromise2;
+
+    expect(item1).toBe(1);
+    expect(item2).toBe(2);
+  });
+
+  it("should handle cancellation", async () => {
```

Comment:
can we also test that the pendingQueue is empty after cancel is called?

Reply:
Sure, great idea!